core-bundle/src/Controller/FaviconController.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\FilesModel;
+use Contao\PageModel;
+use FOS\HttpCache\ResponseTagger;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route(defaults={"_scope" = "frontend"})
+ */
+class FaviconController
+{
+    /**
+     * @var ContaoFramework
+     */
+    private $contaoFramework;
+
+    /**
+     * @var ResponseTagger|null
+     */
+    private $responseTagger;
+
+    public function __construct(ContaoFramework $contaoFramework, ResponseTagger $responseTagger = null)
+    {
+        $this->contaoFramework = $contaoFramework;
+        $this->responseTagger = $responseTagger;
+    }
+
+    /**
+     * @Route("/favicon.ico", name="contao_favicon")
```

Comment:
does the route really need a route name? I mean noone will ever generate that?

Reply:
Imho a route should have a name yes.

Reply:
https://symfony.com/doc/current/routing.html#generating-urls

> If you don't set the route name explicitly with the name option, Symfony generates an automatic name based on the controller and action.

I think that would be totally fine in this case, noone ever will override the named route, as it is not used to generate a route anyway.